src/coreclr/jit/gentree.cpp

```diff
@@ -21017,6 +21017,20 @@ GenTree* Compiler::gtNewSimdBinOpNode(
 
             return gtNewSimdBinOpNode(GT_AND, type, shiftOp, maskOp, simdBaseJitType, simdSize);
         }
+
+        case GT_LSH:
+        {
+            assert(varTypeIsByte(simdBaseType));
+            GenTree* andShiftVal = fgMakeMultiUse(&op2ForLookup);
+            GenTree* shiftOp =
+                gtNewSimdBinOpNode(GT_LSH, type, op1, op2ForLookup,
+                                   varTypeIsUnsigned(simdBaseType) ? CORINFO_TYPE_USHORT : CORINFO_TYPE_SHORT,
+                                   simdSize);
+            GenTree* maskElement =
+                gtNewOperNode(GT_LSH, andShiftVal->TypeGet(), gtNewIconNode(0xFF), fgMakeMultiUse(&andShiftVal));
+            GenTree* mask = gtNewSimdCreateBroadcastNode(type, maskElement, simdBaseJitType, simdSize);
```

Comment:
> I think we could just share the code path even since the code looks to be identical other than using GT_LSH vs GT_RSZ, so we can probably just pass in op there instead.

Yes, I think you're right. I've reworked the new `GT_LSH` logic so it shares the same code path as `GT_RSZ` now.